### PR TITLE
Accept non-integer statusCode JSON values

### PR DESCRIPTION
**Pull request by slank**
*Description of changes:*

The Lambda proxy integration for API Gateway does not specify that statusCode must be an integer, but SAM CLI requires it. This change fixes SAM CLI to accept non-integer status codes that contain integer values.

See https://github.com/slank/awsgi/pull/23 for additional background.

*Checklist:*

- [ ] Write Design Document ([Do I need to write a design document?](https://github.com/awslabs/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.rst#design-document))
- [x] Write unit tests
- [ ] Write/update functional tests
- [ ] Write/update integration tests
- [x] `make pr` passes
- [ ] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/unit/local/apigw/test_local_apigw_service.py [hunk 1]
```diff
@@ -341,6 +341,15 @@ def test_status_code_not_int(self):
                                                    binary_types=[],
                                                    flask_request=Mock())
 
+    def test_status_code_int_str(self):
```
Comment (jfuss) [inline tests/unit/local/apigw/test_local_apigw_service.py]: Can you add a test for when ValueError is raised when status_code is not a positive integer as a string?

Comment (slank) [reply tests/unit/local/apigw/test_local_apigw_service.py]: Sure thing. Added.

**Comment by slank** [comment]
I've added an integration test in what I _think_ is the right place. I get S3 client errors trying to run those tests locally so we'll see what Travis says.

**Comment by jfuss** [comment]
@slank Thanks for adding the tests. Some of our integ-tests require creds, so it's could be either you don't have the correct permissions or no credentials.

**Pull request merged**